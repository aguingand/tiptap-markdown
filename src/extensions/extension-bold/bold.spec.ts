import { expect, test } from "vitest";
import { parse, serialize } from "../../../__tests__/utils";


test('parse markdown with "*"', () => {
    expect(parse('**example**')).toMatchSnapshot();
});
test('parse markdown with "_"', () => {
    expect(parse('__example__')).toMatchSnapshot();
});
test('parse html', () => {
    expect(parse('<strong>example</strong>')).toMatchSnapshot();
});
test.skip('serialize', () => {
    expect(serialize('<strong>example</strong>')).toEqual('**example**');
    expect(serialize('<strong data-markdown-marker="_">example</strong>')).toEqual('__example__');
});
