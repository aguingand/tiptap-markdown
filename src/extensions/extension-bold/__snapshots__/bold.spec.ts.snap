// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse html 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "data-markdown-marker": "*",
            },
            "type": "bold",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown with "*" 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "data-markdown-marker": "*",
            },
            "type": "bold",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`parse markdown with "_" 1`] = `
[
  {
    "content": [
      {
        "marks": [
          {
            "attrs": {
              "data-markdown-marker": "_",
            },
            "type": "bold",
          },
        ],
        "text": "example",
        "type": "text",
      },
    ],
    "type": "paragraph",
  },
]
`;
